### DISCUSSION TITLE
Precipitation from observations

### DISCUSSION THREAD

**Discussion by jsminor16** [weather-gov/api#622] [Q&A]
The precipitation values from the api are different than the ones on the website. It almost seems like an integer is being passed instead of a float? I either get 0 or nan for values. The zeroes line up with the hourly report of rain on the website. nan's line up with the hours when there wasn't any rain.

**Comment by sullynole** [comment]
Are the values less than 0.4"?  There is a known issue mentioned on the updates tab here: https://www.weather.gov/documentation/services-web-api

**Reply by sthiboldeaux** [reply]
I am noticing a similar issue, but it seems like the precip values are rounding to the nearest cm integer. This is 0 in many cases, but all other values are rounded integers (and shows up as multiples of 10 in mm).

https://github.com/weather-gov/api/discussions/645

**Reply by sullynole** [reply]
That aligns with the known issue, which is a precision bug with ingest of precipitation values to API backend from MADIS.

**Reply by shucktech** [reply]
Stumbled on this thread after digging into why most the `precipitationLast3Hours` (if not all) was always showing either 0 or `None` - looks like I found my reason:

> The observations endpoints always show missing (null) wind gust and may show null values for other weather properties (temperature, wind, precipitation, etc.) at stations when the [MADIS source](https://madis-data.ncep.noaa.gov/) has data.
Precipitation values in the observations endpoints are rounded down to the nearest centimeter (less than 0.4" may be improperly rounded down to 0).

That's a bummer. Anyone know of an alternative? Or another API endpoint that'll offer a similar metric?

**Comment by LorenzoPeve** [comment]
This issue still exists @nickmachairas.

**Comment by dangelsaurus** [comment]
Curious if anyone found another place to get precipitation data?  Right now I'm relying on forecast data as a proxy for estimated rainfall, but would love observational data without having to manually pull up each station. https://www.weather.gov/wrh/timeseries?site=Kdfw

**Reply by sullynole** [reply]
RFC QPE: https://water.noaa.gov/about/precipitation-data-access

Also can be found in RTMA/URMA: https://www.nco.ncep.noaa.gov/pmb/products/rtma/

**Comment by jah1007** [comment]
The release to fix this bug is tentatively scheduled to be activated next Tuesday. Precipitation data is now be available on preview-api.weather.gov for testing.

**Reply by jah1007** [reply]
We were delayed by the flooding in the Pacific Northwest. The update has been pushed to production.
